### PR TITLE
Updating the ReadGain functions to use the dB gain from Xf files, the…

### PR DESCRIPTION
**Pull request by FlahertyJustinC**
…n convert ot linear gain.  This was done to preserve significant figures, as the XF conversion to linear gain only goes up to two decimal places.

### CODE REVIEW

**Comment by FlahertyJustinC** [comment]
I should also note that this required a slight update to the Veff test, as the updated sig figs change the total weight of the test in the thousandths decimal place.

File: test/veff/check_veff.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	And summed weight passing
 '''
 expected_global_pass = 2 #Updated this from 3 after changing the effective height calculation and number of events thrown - JCF 2/23/20242
```
Comment (abigailbishop) [inline test/veff/check_veff.py]: Idk if this comment is necessary given the information should be stored in the Git Blame right? Same with the comment in the line below

File: test/veff/setup_veff_test.txt [hunk 2]
```diff
@@ -39,3 +39,5 @@ RAYSOL_RANGE=9000 // radius of the cylinder in ray tracer will look for solution
 //the output
 DATA_SAVE_MODE=2 // 2: save only physics data
 DATA_LIKE_OUTPUT=0 //don't try to save anything to event tree
+
+FILL_TREE_MODE=0
```
Comment (abigailbishop) [inline test/veff/setup_veff_test.txt]: Does this still need to be in mode 0? I thought that change was just for debugging and idk the size limitations of the GitHub tests.

File: Report.cc [hunk 3]
```diff
@@ -1929,14 +1928,15 @@ void Report::Connect_Interaction_Detector_V2(Event *event, Detector *detector, R
                                                 //End Fresnel application
 
                                                 //Apply birefringence.
-                                                birefringence->Principal_axes_polarization(Pol_vector, bire_ray_cnt, max_bire_ray_cnt, settings1); //For birefringence, modify the polarization at the antennas                                            
+                                                birefringence->Principal_axes_polarization(Pol_vector, bire_ray_cnt, max_bire_ray_cnt, settings1); //For birefringence, modify the polarization at the antennas       
+                                                
+                                                //Debugging
+                                                int testN = 100;
```
Comment (abigailbishop) [inline Report.cc]: Ik it's annoying to change back and forth, but if this is for debugging does it need to be pushed to the main branch? We don't want to push too much debugging stuff to the main branch that way it never gets too overwhelming.

File: Report.cc [hunk 4]
```diff
@@ -5059,9 +5094,9 @@ double Report::GaintoHeight(double gain, double freq, double n_medium, double Z_
     // and h_eff=2*sqrt(A_eff*Z_rx/Z_air)
     // gain is unitless value
     
-    // return sqrt(gain/4/PI*CLIGHT*CLIGHT/(freq*freq*n_medium*n_medium)*Z_A/(Z0/n_medium));  // n_medium parts are changed from icemc(I believe this is correct one; E. Hong)
+    return sqrt(gain/4/PI*CLIGHT*CLIGHT/(freq*freq*n_medium*n_medium)*Z_A/(Z0/n_medium));  // n_medium parts are changed from icemc(I believe this is correct one; E. Hong)
```
Comment (abigailbishop) [inline Report.cc]: This is minutia, but since you visited this line could you add some spaces to improve the readability of this expression?

File: Report.cc [hunk 5]
```diff
@@ -2071,12 +2072,20 @@ void Report::Connect_Interaction_Detector_V2(Event *event, Detector *detector, R
...
                                                 	{
                                                     		ApplyElect_Tdomain_FirstTwo(freq_tmp *1.e-6, freq_lastbin *1.e-6, detector, V_forfft[2 *n], V_forfft[2 *n + 1], gain_ch_no);
-                                                	}
+                                                	}                                                   
 
                                             	}   // end for freq bin
                                                 //End amplification at receiving antenna
                                                 
-
+                                                //Debugging print statements
```
Comment (marcomuzio) [inline Report.cc]: If this commented section is no longer needed we should probably remove it to avoid clutter.

File: Report.cc [hunk 6]
```diff
@@ -5140,10 +5141,17 @@ void Report::ApplyAntFactors_Tdomain (double AntPhase, double heff, Vector &n_tr
...
         }
-        if(useInTransmitterMode==true){ phase_current += PI/2;};
-        // V amplitude
-        double v_amp  = sqrt(vm_real*vm_real + vm_img*vm_img) / sqrt(2.) * 0.5 * heff * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
-        if(useInTransmitterMode==true){ v_amp *= freq/CLIGHT*(2*Z0/(Zr + Z_A));};
+        double v_amp  = sqrt(vm_real*vm_real + vm_img*vm_img);
+        if (useInTransmitterMode==true){ 
+            phase_current += PI/2;
+            v_amp *= heff * pol_factor;
+            // v_amp *= freq/CLIGHT*(2*Z0/Zr);// I might need to reduce this by a factor of two. - JCF 2/25/2024
```
Comment (marcomuzio) [inline Report.cc]: Can you updated the comments here and keep only the line that you settled on?

File: Report.cc [hunk 7]
```diff
@@ -5140,10 +5141,17 @@ void Report::ApplyAntFactors_Tdomain (double AntPhase, double heff, Vector &n_tr
...
+            v_amp *= freq/CLIGHT*(Z0/Zr)/4/sqrt(2.);// Testing factor of two reduction - JCF 2/27/2024           
+        }
+        else {
+            // V amplitude
+            v_amp *= 1 / sqrt(2.) * 0.5 * heff * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
```
Comment (marcomuzio) [inline Report.cc:5142]: Did we need an if statement here for the different stations? Or maybe you wanna do that in a separate PR.

File: Report.cc [hunk 8]
```diff
@@ -5162,12 +5170,18 @@ void Report::ApplyAntFactors_Tdomain_FirstTwo (double heff, double heff_lastbin,
...
     if (useInTransmitterMode) {
-        vm_bin0 *= freq/CLIGHT*(2*Z0/(Zr + Z_A));
-        vm_bin1 *= freq/CLIGHT*(2*Z0/(Zr + Z_A));
+        // The factors of two are currently being explored in this. - JCF 2/29/2024
+        vm_bin0 *= freq/CLIGHT*(Z0/(Zr))/4/sqrt(2.);
+        vm_bin1 *= freq/CLIGHT*(Z0/(Zr))/4/sqrt(2.);
+    }
+    else {
+        vm_bin0 *= 1 /sqrt(2.) * 0.5 * heff * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
+        vm_bin1 *= 1 / sqrt(2.) * 0.5 * heff_lastbin * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
```
Comment (marcomuzio) [inline Report.cc:5173]: Same question about the factors of sqrt(2) here.

File: Report.cc [hunk 9]
```diff
@@ -5162,12 +5170,18 @@ void Report::ApplyAntFactors_Tdomain_FirstTwo (double heff, double heff_lastbin,
...
-    vm_bin0 = vm_bin0 / sqrt(2.) * 0.5 * heff * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
-    vm_bin1 = vm_bin1 / sqrt(2.) * 0.5 * heff_lastbin * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
+    //First step in the voltage calculation.  Both Tx and Rx mode use this.
+    vm_bin0 *= heff * pol_factor;
+    vm_bin1 *= heff_lastbin * pol_factor;
     
     if (useInTransmitterMode) {
-        vm_bin0 *= freq/CLIGHT*(2*Z0/(Zr + Z_A));
-        vm_bin1 *= freq/CLIGHT*(2*Z0/(Zr + Z_A));
+        // The factors of two are currently being explored in this. - JCF 2/29/2024
```
Comment (marcomuzio) [inline Report.cc]: I think this comment will definitely be a mystery in a year's time. Probably better to remove it?

File: Report.cc [hunk 10]
```diff
@@ -5255,14 +5240,7 @@ void Report::InvertAntFactors_Tdomain_FirstTwo (double heff, double heff_lastbin
...
     vm_bin0 = vm_bin0 / sqrt(2.) * 0.5 * heff * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
-    vm_bin1 = vm_bin1 / sqrt(2.) * 0.5 * heff_lastbin * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
-    
-    // if (std::isnan(vm_bin0)) {
-    //     vm_bin0 = 0.0;
-    // }
-    // if (std::isnan(vm_bin1)) {
-    //     vm_bin1 = 0.0;      
-    // }    
+    vm_bin1 = vm_bin1 / sqrt(2.) * 0.5 * heff_lastbin * pol_factor; // sqrt(2) for 3dB splitter for TURF, SURF, 0.5 to calculate power with heff
```
Comment (marcomuzio) [inline Report.cc:5232]: Here's another spot with the sqrt(2). Apologies for the spam if it's being handled later.

**Comment by marcomuzio** [comment]
Hey @jflaherty13 I think this looks good overall. I just did a quick skim of the code and didn't check the math in any detail, but I'm sure that part is working since you've tested this pretty thoroughly. Could you update the log.txt to describe your new pulser sims and updates to the antenna gain & impedances?

File: Report.cc (hunk 7 above)
Comment (FlahertyJustinC) [reply Report.cc:5142]: That'll be a separate PR.  The splitters seem to be included in the electronics gain files, so we will likely remove the sqrt(2) completely.  We want to carefully check this before we bump our effective heights back up though.

File: Report.cc (hunk 5 above)
Comment (FlahertyJustinC) [reply Report.cc]: Removed

File: Report.cc (hunk 6 above)
Comment (FlahertyJustinC) [reply Report.cc]: I'll reduce it to one line, but I'm leaving a comment that these factors of two are still up for debate.

Comment (marcomuzio) [reply Report.cc]: Thanks!

File: Report.cc (hunk 7 above)
Comment (marcomuzio) [reply Report.cc:5142]: Sounds good to me!
